exec/Chainweb/BackfillTransfers.hs

```diff
@@ -0,0 +1,330 @@
+      Just (A.Number n) -> toRealFloat n
+      _ -> case _ev_params ev ^? to unwrap . ix 2 . key "int" of
+        Just (A.Number n) -> toRealFloat n
+        _ -> maybe (error $ msg "amount") toRealFloat $ _ev_params ev ^? to unwrap . ix 2 . _Number
+    msg :: String -> String
+    msg = printf "Chainweb.BackfillTransfers.createTrantsfer: This entry (%s) was not found"
+    from_acct ev = fromMaybe (error $ msg "from_acct") $ _ev_params ev ^? to unwrap . ix 0 . _String
+    to_acct ev = fromMaybe (error $ msg "to_acct") $ _ev_params ev ^? to unwrap . ix 1 . _String
+    unwrap (PgJSONB a) = a
+    go ev creationTime = Transfer
+      {
+        _tr_creationtime = creationTime
+      , _tr_block = _ev_block ev
+      , _tr_requestkey = _ev_requestkey ev
+      , _tr_chainid = _ev_chainid ev
+      , _tr_height = _ev_height ev
+      , _tr_idx = _ev_idx ev
+      , _tr_qualName = _ev_qualName ev
+      , _tr_modulename = _ev_module ev
+      , _tr_moduleHash = _ev_moduleHash ev
+      , _tr_from_acct = from_acct ev
+      , _tr_to_acct = to_acct ev
+      , _tr_amount = amount ev
+      }
+
+splitEvents :: [Event] -> ([Event],[Event])
+splitEvents = partitionEithers . map split
+  where
+    split ev = case _ev_requestkey ev of
+      RKCB_RequestKey _ -> Left ev
+      RKCB_Coinbase -> Right ev
+
+-- including the bounding heights triggers index usage (specifically the transactions_height_idx index)
+creationTimesByRequestKeysNonCoinbase :: Int64 -> Int64 -> Int64 -> [Event] -> Q Postgres ChainwebDataDb s (C (QExpr Postgres s) UTCTime)
+creationTimesByRequestKeysNonCoinbase minHeight maxHeight cid evs = do
+    t <- all_ (_cddb_transactions database)
+    guard_ $ _tx_chainId t ==. val_ cid
+    guard_ $ _tx_height t <. val_ maxHeight
+    guard_ $ _tx_height t >=. val_ minHeight
+    guard_ $ _tx_requestKey t `in_` (val_ <$> (getDbHashes $ map _ev_requestkey evs))
+    return $ _tx_creationTime t
+  where
+    getDbHashes = mapMaybe $ \case
+      RKCB_RequestKey t -> Just t
+      _ -> Nothing
+
+creationTimesByRequestKeysCoinbase :: Int64 -> Int64 -> Int64 -> [Event] -> Q Postgres ChainwebDataDb s (C (QExpr Postgres s) UTCTime)
+creationTimesByRequestKeysCoinbase minHeight maxHeight cid evs = do
+    t <- all_ (_cddb_transactions database)
+    guard_ $ _tx_height t <. val_ maxHeight
+    guard_ $ _tx_height t >=. val_ minHeight
+    guard_ $ _tx_chainId t ==. val_ cid
+    guard_ $ _tx_block t `in_` (val_ <$> (getBlockHashes $ map (\ev -> (_ev_block ev, _ev_requestkey ev)) evs))
+    return $ _tx_creationTime t
+  where
+    getBlockHashes  = mapMaybe $ \case
+      (bhash, RKCB_Coinbase) -> Just bhash
+      _ -> Nothing
+
+eventSelector :: Int64 -> Int64 -> Maybe Int64 -> Q Postgres ChainwebDataDb s (EventT (QExpr Postgres s))
+eventSelector startingHeight endingHeight chainid = do
+      ev <- all_ (_cddb_events database)
+      guard_ $ _ev_height ev <. val_ endingHeight
```

Comment:
I think this should be `<=.` based on the reasoning that we may not have gotten all the entries at `endingHeight` on the previous pass and having a small number of multiple inserts shouldn't be a problem.